docs/lib/introduction.md

```diff
@@ -0,0 +1,642 @@
+
+    type MessageRegistry map[string]Message
+
+Typically, the [package defining the type registry](introduction/prototype-based-typeregistry/registry/registry.go) also
+declares and
+initializes a _DefaultRegistry_, a global variable that can be used by other packages to register their implementation
+of the respective type (here, Message).
+
+    var DefaultMessageRegistry = MessageRegistry{}
+
+To register themselves at the _DefaultRegistry_, these other packages implement their
+[init-function](https://go.dev/doc/effective_go#init) respectively (thereby, the type name should obviously be unique).
+
+    const TYPE = "simplemessage"
+
+    func init() {
+        registry.DefaultMessageRegistry[TYPE] = &Message{}
+    }
+    
+    type Message struct {
+        Type string `json:"type"`
+        Text string `json:"text"`
+    }
+    
+    func (m *Message) Print() {
+        fmt.Println(m.Text)
+    }
+
+Keep in mind that although the
+[_simplemessage_](introduction/prototype-based-typeregistry/types/simplemessage/message.go) and
+[_complexmessage_](introduction/prototype-based-typeregistry/types/complexmessage/message.go) packages might not be
+directly used in other packages, they have to be imported for their side effects (in other words, to execute their
+init-functions). You may use empty imports to achieve this.
+
+    import (
+        _ "example/typeregistry/types/complexmessage"
+        _ "example/typeregistry/types/simplemessage"
+    )
+
+Now, the prepared _DefaultRegistry_ can be used to dynamically unmarshal serializations of the types into their
+respective go types.
+
+The [registry](introduction/prototype-based-typeregistry/registry/registry.go) has a function implementing the dynamic
+unmarshaling logic. To understand this, it's best to check out the well-documented method within the example.
+
+#### Section 2.1.2 - Factory-Based Type Registry
+
+In _Factory-Based Type Registries_, the Registry is essentially a map that stores the model type name (key) and a
+factory object (thus, an object capable of producing Message objects). Thereby, it is possible to overcome the
+limitations of the Prototype-Based Type Registries (the necessity of a dedicated go type per model type and not being
+able to edit the object after unmarshaling).
+
+Based on the knowledge about Prototype-Based Type Registry, it should be possible to understand the well-documented
+[Factory-Based Type Registry](introduction/factory-based-typeregistry) example.
+
+#### Note
+
+The ocm-lib uses has several kinds of type registries (thus, corresponding to our example, not only a Message Registry
+but also for other types). There, it has a generic implementation of a registry which can be reused.  
+Furthermore, its implementation of the registry has to be thread safe and provides some convenience functionility.
+Therefore, it may be slightly more difficult to understand than our examples.
+
+#### Section 2.1.3 - Purpose
```

Comment:
I propose to move this "Purpose" Section up, maybe include this directly under "Section 2.1 - (Model) Type Registries". It is crucial to first of all understand (and therefore motivate the reader) _why_ these complex type registries are required in the OCM Lib, and only then talk about the _how_